homeassistant/components/onvif/__init__.py

```diff
@@ -60,8 +63,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     platforms = ["camera"]
 
     if device.capabilities.events and device.events.started:
-        platforms += ["binary_sensor", "sensor"]
+        platforms += [Platform.BINARY_SENSOR, Platform.SENSOR]
         await device.events.async_stop()
+    if device.capabilities.imaging:
+        platforms += [Platform.SENSOR]
```

Comment:
Should this be a switch platform as that's what we added for imaging when setting up the config entry?

```suggestion
        platforms += [Platform.SWITCH]
```